packages/arb-provider-ethers/src/lib/wallet.ts

```diff
@@ -166,17 +178,77 @@ export class ArbWallet extends ethers.Signer {
     const vmId = await this.provider.getVmID()
     const valueNum = ethers.utils.bigNumberify(value)
     const globalInbox = await this.globalInboxConn()
-    const seq = await this.generateSeq()
-    const tx = await globalInbox.sendTransactionMessage(
-      vmId,
-      to,
-      seq,
-      valueNum,
-      data
-    )
-    const tx2 = this.provider._wrapTransaction(tx, tx.hash)
-    await this.incrementSeq()
-    return tx2
+    const seq = await this.generateAndIncrementSeq()
+    const from = await this.getAddress()
+
+    try {
+      if (this.provider.aggregator) {
+        const arbTxHash = Hashing.calculateTransactionHash(
+          vmId,
+          to,
+          from,
+          ethers.utils.bigNumberify(seq),
+          valueNum,
+          data
+        )
+
+        const offchainTxHash = Hashing.calculateOffchainTransactionHash(
```

Comment:
Makes sense, updated.

Reply:
When we implement channels, we're going to want "offchain Tx" to mean something else.  So I would recommend a name other than "offchain Tx" here.  Maybe include "batch" in the name?

Reply:
Again this code had actually already been changed, but I just updated it slightly again, I decided to use batch in the name rather than aggregator to keep the naming scheme in line with the global inbox function. An aggregator is one possible role that submits batches.